articles/greedy-algorithms/index.md

```diff
@@ -0,0 +1,159 @@
+        raise ValueError('Getting a profit requires at least 2 prices')
+    
+    min_price = price_list[0]
+    max_profit = price_list[1] - price_list[0]
+    
+    for day in range(1, len(price_list)):
+        stock_price_current = price_list[day]
+        tentative_profit = stock_price_current - min_price
+        max_profit = max(max_profit, tentative_profit)
+        min_price = min(min_price, stock_price_current)
+    
+    return max_profit
+
+# inputs
+price1 = [1, 5, 3, 2]
+price2 = [7, 2, 8, 9]
+price3 = [1, 6, 7, 9]
+price4 = [9, 7, 4, 1]
+price5 = [10, 5, 1, 0]
+price6 = [100, 180, 260, 310, 40, 535, 695]
+
+print(" List"," --> Profit(₹)")
+
+print("1.", price1, "--> ₹", maximize_profit(price1))
+print("2.",price2,"--> ₹",maximize_profit(price2))
+print("3.",price3,"--> ₹",maximize_profit(price3))
+print("4.",price4,"--> ₹",maximize_profit(price4))
+print("5.",price5,"--> ₹",maximize_profit(price5))
+print("6.",price6,"--> ₹",maximize_profit(price6))
+```
+
+
+### Output
+
+The output of the code is given below:
+
+```bash
+     List      --> Profit(₹)
+1. [1, 5, 3, 2] --> ₹ 4
+2. [7, 2, 8, 9] --> ₹ 7
+3. [1, 6, 7, 9] --> ₹ 8
+4. [9, 7, 4, 1] --> ₹ -2
+5. [10, 5, 1, 0] --> ₹ -1
+6. [100, 180, 260, 310, 40, 535, 695] --> ₹ 655
+```
+
+### Output Analysis
+
+Let us analyze the output obtained. 
+
+#### Case 1:
+*Input List*: [1,5,3,2]
+
+*Output*: **Maximum Profit: 4**: **Buy at 1** : **Sell at 5**
+
+#### Case 2:
+
+*Input List*: [9,7,4,1]
+
+*Output*: **Maximum Profit: -2**: **Buy at 9** : **Sell at 7**
+
+#### Case 3:
+
+Input: Large numbers, more extensive list
+
+*Input List*: [100,180,260,310,40,535,695]
+*Output*: **Maximum Profit: 655** : Buy at 40 : Sell at 695
+
+##### Optimal Solution:
+- Buy at 100, Sell at 310
+- Buy at 40, Sell at 695
+
+**Total Profit = 865**
+
+
+### Advantages of Greedy Approach
+
+- The worst-case time complexity of the function `maximize_profit()` is Θ(n).
+- Space Complexity of the function is Θ(1).
+- The program completes execution within one pass of the entire list
+- Since it uses a greedy approach, the profits are added up in each step, thereby ensuring profit
+  
+### Limitations of Greedy Approach
+
+- The program can work with only one buy and one sell operation due to its greedy nature. It fails to sell and buy at multiple time stamps.
+- Greedy algorithm solutions are not always optimal. Therefore, the maximum profit computed may be a local maximum. The program can fail to reach the global maxima. For example, the optimal solution in scenario-3 is 865. The output of the greedy algorithm is 655.
+
+### Conclusion
+
+In this article, we considered the greedy approach to obtain the maximum profit given a list of indices. Observe that the option of performing multiple buys and sell operations are never used. I hope the intuition behind the greedy approach was presented well. Do let me know what you think about the approach taken to solve the problem. Feedback is highly appreciated.
```

Comment:
This approach to Greedy Algorithms were excellent, and well presented. 
It is very straightforward and easy for beginners to understand. This article could be improved by including and discussing other approaches in brief. A part 2 with a detailed discussion could compliment the article better. 

Make the suggested changes and we can move on to the next phase of review, @lalith1403.